### PR TITLE
Implement np.linalg.eig on CPU.

### LINKED ISSUE #639: Feature request for computing eigenvalues of non-hermitian matrix.
I'm basically looking for np.linalg.eig to be implemented in jax or jax.lax.  I personally am looking for a speedup in the computation, and am not interested the derivative at this time.
Thanks!

### PR DESCRIPTION
**Pull request by hawkinsp**
Fixes #639.

### CODE REVIEW

File: jax/lax_linalg.py [hunk 1]
```diff
@@ -117,6 +121,48 @@ def cholesky_cpu_translation_rule(c, operand):
 
 xla.backend_specific_translations['cpu'][cholesky_p] = cholesky_cpu_translation_rule
 
+# Asymmetric eigendecomposition
+
+def eig_impl(operand):
+  w, vl, vr = xla.apply_primitive(eig_p, operand)
+  return core.pack((w, vl, vr))
```
Comment (mattjj) [inline jax/lax_linalg.py]: I believe you can shorten this to just

```
return xla.apply_primitive(eig_p, operand)
```

That is, the application of the primitive will return a JaxTuple (or a DeviceTuple, which is to a JaxTuple what a DeviceArray is to a numpy.ndarray), so there's no need to unpack it and then re-pack it.

File: jax/lax_linalg.py [hunk 2]
```diff
@@ -117,6 +121,48 @@ def cholesky_cpu_translation_rule(c, operand):
...
+
+def eig_translation_rule(c, operand):
+  raise NotImplementedError(
+    "Nonsymmetric eigendecomposition is only implemented on the CPU backend")
+
+def eig_abstract_eval(operand):
+  if isinstance(operand, ShapedArray):
+    if operand.ndim < 2 or operand.shape[-2] != operand.shape[-1]:
+      raise ValueError("Argument to nonsymmetric eigendecomposition must have "
+                       "shape [..., n, n]")
```
Comment (mattjj) [inline jax/lax_linalg.py]: Could report what operand.shape is here.

File: tests/linalg_test.py [hunk 3]
```diff
@@ -104,6 +104,48 @@ def testSlogdet(self, n, dtype, rng):
...
+       "n": n, "dtype": dtype, "rng": rng}
+      for n in [0, 4, 5, 50]
+      for dtype in float_types() + complex_types()
+      for rng in [jtu.rand_default()]))
+  # TODO(phawkins): enable when there is an eigendecomposition implementation
+  # for GPU/TPU.
+  @jtu.skip_on_devices("gpu", "tpu")
+  def testEig(self, n, dtype, rng):
+    self.skipTest("Test disabled until Jaxlib 0.1.15 is released") # TODO(phawkins)
+    args_maker = lambda: [rng((n, n), dtype)]
```
Comment (mattjj) [inline tests/linalg_test.py]: Could test a leading dimension here too, right? (Since NumPy supports arg shapes of `(..., M, M)`.)

**Review by mattjj — APPROVED** [review]
Beautiful!

I didn't check the cython code very closely, but the test coverage proves it works.

**Comment by hawkinsp** [comment]
All done.
